File: app/src/main/java/jp/ergo/android/screentouchdispatcher/NotificationUtils.java
package jp.ergo.android.screentouchdispatcher;

import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.support.v4.app.NotificationManagerCompat;
import android.support.v7.app.NotificationCompat;

/**
 * Created by ishigayawataru on 15/12/31.
 */
public class NotificationUtils {

    public static final int NOTIFICATION_ID = 1;


    public static Intent createBroadcastIntent(final Context context) {
        final Intent broadcastIntent = new Intent(context, ScreenTouchDispatcherBroadcastReceiver.class);
        final String action = ScreenTouchDispatcherBroadcastReceiver.ACTION_SWITCH;
        broadcastIntent.setAction(action);
        return broadcastIntent;
    }
    public static void removeNotification(final Context context) {
        NotificationManagerCompat manager = NotificationManagerCompat.from(context);
        manager.cancel(NotificationUtils.NOTIFICATION_ID);
    }


    public static void sendNotification(final Context context, final String message) {
        NotificationCompat.Builder builder = new NotificationCompat.Builder(context);

        final Intent broadcastIntent = NotificationUtils.createBroadcastIntent(context);
        final PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 100, broadcastIntent, PendingIntent.FLAG_UPDATE_CURRENT);

        builder.setContentIntent(pendingIntent);
        builder.setSmallIcon(R.mipmap.ic_launcher);
        builder.setContentText(message);
        builder.setOngoing(true);

        NotificationManagerCompat manager = NotificationManagerCompat.from(context);
        manager.notify(NotificationUtils.NOTIFICATION_ID, builder.build());

    }

}

File: app/src/main/kotlin/jp/ergo/android/screentouchdispatcher/NotificationUtils.kt
package jp.ergo.android.screentouchdispatcher

import android.app.Notification
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.support.v4.app.NotificationManagerCompat
import android.support.v7.app.NotificationCompat

object NotificationUtils {

    val NOTIFICATION_ID = 1

    fun createBroadcastIntent(context: Context): Intent {
        val broadcastIntent = Intent(context, ScreenTouchDispatcherService::class.java)
        val action = ScreenTouchDispatcherService.ACTION_SWITCH
        broadcastIntent.action = action
        return broadcastIntent
    }

    fun removeNotification(context: Context) {
        val manager = NotificationManagerCompat.from(context)
        manager.cancel(NotificationUtils.NOTIFICATION_ID)
    }


    fun sendNotification(context: Context, message: String) {
        val manager = NotificationManagerCompat.from(context)
        manager.notify(NotificationUtils.NOTIFICATION_ID, createNotification(context, message))
    }

    fun createNotification(context: Context, message: String): Notification {
        val builder = NotificationCompat.Builder(context)

        val broadcastIntent = NotificationUtils.createBroadcastIntent(context)
        val pendingIntent = PendingIntent.getService(context, 100, broadcastIntent, PendingIntent.FLAG_UPDATE_CURRENT)

        builder.setContentIntent(pendingIntent)
        builder.setSmallIcon(R.mipmap.ic_launcher)
        builder.setContentText(message)
        builder.setOngoing(true)
        return builder.build()
    }

}
